### PR TITLE
Replaces index_update with .at().set()

### PR DESCRIPTION
**Pull request by badisa**
* index_update was deprecated in 0.2.22 (https://jax.readthedocs.io/en/latest/changelog.html#jax-0-2-22-oct-12-2021)
* Originally identified by James Thompson, provided initial patches

### CODE REVIEW

**Comment by proteneer** [comment]
whoa! I had no idea they deprecated this.

**Comment by proteneer** [comment]
its a little annoying that `.set()` returns an object as semantically it seems to mean "set in place" in my head.

**Review by proteneer — APPROVED** [review]

File: timemachine/fe/topology.py [hunk 1]
```diff
@@ -585,13 +584,13 @@ def parameterize_nonbonded(self, ff_q_params, ff_lj_params):
...
-        src_qlj_params_a = jax.ops.index_update(src_qlj_params_a, jax.ops.index[:, 2], src_qlj_params_a[:, 2] * 0.5)
+        src_qlj_params_a = jnp.asarray(qlj_params_a).at[:, 0].set(qlj_params_a[:, 0] * 0.5)
+        src_qlj_params_a = jnp.asarray(src_qlj_params_a).at[:, 2].set(src_qlj_params_a[:, 2] * 0.5)
         dst_qlj_params_a = qlj_params_a
 
         qlj_params_b = standard_qlj_typer(self.mol_b)
-        src_qlj_params_b = jax.ops.index_update(qlj_params_b, jax.ops.index[:, 0], qlj_params_b[:, 0] * 0.5)
-        src_qlj_params_b = jax.ops.index_update(src_qlj_params_b, jax.ops.index[:, 2], src_qlj_params_b[:, 2] * 0.5)
+        src_qlj_params_b = jnp.asarray(qlj_params_b).at[:, 0].set(qlj_params_b[:, 0] * 0.5)
+        src_qlj_params_b = jnp.asarray(src_qlj_params_b).at[:, 2].set(src_qlj_params_b[:, 2] * 0.5)
```
Comment (maxentile) [inline timemachine/fe/topology.py:593]: nit: these can be simplified slightly `x.at[idx].set(x[idx] * 0.5)` -> `x.at[idx].multiply(0.5)`

**Review by maxentile — APPROVED** [review]
Looks good!

**Review by mcwitt — APPROVED** [review]

**Pull request merged**